pkg/storage/unified/testing/storage_backend.go

```diff
@@ -471,15 +470,81 @@ func runTestIntegrationBackendList(t *testing.T, backend resource.StorageBackend
 		require.NoError(t, err)
 		require.Nil(t, res.Error)
 		require.Len(t, res.Items, 2)
-		t.Log(res.Items)
-		require.Contains(t, string(res.Items[0].Value), "item4 ADDED")
-		require.Contains(t, string(res.Items[1].Value), "item5 ADDED")
+		require.Contains(t, string(res.Items[0].Value), "item5 ADDED")
+		require.Contains(t, string(res.Items[1].Value), "item2 MODIFIED")
 
 		continueToken, err = resource.GetContinueToken(res.NextPageToken)
 		require.NoError(t, err)
 		require.Equal(t, rv8, continueToken.ResourceVersion)
 		require.Equal(t, int64(4), continueToken.StartOffset)
 	})
+
+	t.Run("Paginate through latest items one by one", func(t *testing.T) {
+		baseKey := &resourcepb.ResourceKey{
+			Namespace: ns,
+			Group:     "group",
+			Resource:  "resource",
+		}
+		expectedItems := []string{"item1 ADDED", "item4 ADDED", "item5 ADDED", "item2 MODIFIED", "item6 ADDED"}
+		var allItems []*resourcepb.ResourceWrapper
+		var nextPageToken string
+
+		for i, expectedValue := range expectedItems {
+			req := &resourcepb.ListRequest{
+				Limit:         1,
+				NextPageToken: nextPageToken,
+				Options: &resourcepb.ListOptions{
+					Key: baseKey,
+				},
+			}
+
+			res, err := server.List(ctx, req)
+			require.NoError(t, err)
+			require.Nil(t, res.Error)
+			require.Len(t, res.Items, 1)
+			require.Contains(t, string(res.Items[0].Value), expectedValue)
+			allItems = append(allItems, res.Items[0])
+
+			if i < len(expectedItems)-1 {
+				require.NotEmpty(t, res.NextPageToken, "should have a continue token for page %d", i+1)
+				nextPageToken = res.NextPageToken
+			} else {
+				require.Empty(t, res.NextPageToken, "should not have a continue token on the last page")
+			}
+		}
+		require.Len(t, allItems, len(expectedItems))
+	})
+
+	t.Run("Paginate latest with a limit larger than remaining items", func(t *testing.T) {
+		baseKey := &resourcepb.ResourceKey{
+			Namespace: ns,
+			Group:     "group",
+			Resource:  "resource",
+		}
+		// Request first 3 items (out of 5 total)
+		req := &resourcepb.ListRequest{
+			Limit: 3,
+			Options: &resourcepb.ListOptions{
+				Key: baseKey,
+			},
+		}
+		res1, err := server.List(ctx, req)
+		require.NoError(t, err)
+		require.Nil(t, res1.Error)
+		require.Len(t, res1.Items, 3)
```

Comment:
could we add a `require.contains` for item1, item4 and item5 like we did with block below?